packages/lib/hooks/useKeyPress.ts

```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect, RefObject, useRef } from "react";
+
+export function useKeyPress(targetKey: string, ref?: RefObject<HTMLInputElement>): boolean {
+  // State for keeping track of whether key is pressed
+  const [keyPressed, setKeyPressed] = useState(false);
+  const placeHolderRef = ref?.current;
+  // If pressed key is our target key then set to true
+  function downHandler({ key }: { key: string }): void {
+    if (key === targetKey) {
+      setKeyPressed(true);
+    }
+  }
+  // If released key is our target key then set to false
+  const upHandler = ({ key }: { key: string }): void => {
+    if (key === targetKey) {
+      setKeyPressed(false);
+    }
+  };
+  // Add event listeners
+  useEffect(() => {
+    if (ref && placeHolderRef) {
```

Comment:
I did try this approach but TS wasnt happy :D

Reply:
Hmm, tried it right now myself as well, but indeed somehow the types of the `addEventListener` function change and starts trowing errors. Let's leave it like this then since the file is pretty small 😁